src/System.IO.FileSystem/tests/Directory/GetFileSystemEntries_str_str.cs

```diff
@@ -3,503 +3,171 @@
+        testDir.CreateSubdirectory("TestDir3");
+        using (new FileInfo(Path.Combine(TestDirectory, "TestFile1")).Create())
+        using (new FileInfo(Path.Combine(TestDirectory, "TestFile2")).Create())
+        using (new FileInfo(Path.Combine(TestDirectory, "Test1File2")).Create())
+        using (new FileInfo(Path.Combine(TestDirectory, "Test1Dir2")).Create())
+        {
+            String[] strArr = GetEntries(TestDirectory, "*2");
+            Assert.Equal(4, strArr.Length);
+            Assert.Contains(Path.Combine(TestDirectory, "Test1Dir2"), strArr);
+            Assert.Contains(Path.Combine(TestDirectory, "TestDir2"), strArr);
+            Assert.Contains(Path.Combine(TestDirectory, "Test1File2"), strArr);
+            Assert.Contains(Path.Combine(TestDirectory, "TestFile2"), strArr);
+
+            strArr = GetEntries(TestDirectory, "*Dir2");
+            Assert.Equal(2, strArr.Length);
+            Assert.Contains(Path.Combine(TestDirectory, "Test1Dir2"), strArr);
+            Assert.Contains(Path.Combine(TestDirectory, "TestDir2"), strArr);
+        }
+    }
 
+    [Fact]
+    public void SearchPatternCaseSensitivity()
+    {
+        DirectoryInfo testDir = new DirectoryInfo(TestDirectory);
+        using (new FileInfo(Path.Combine(TestDirectory, "AAABB")).Create())
+        {
+            Directory.CreateDirectory(Path.Combine(TestDirectory, "aaabbcc"));
+            String[] strArr = GetEntries(TestDirectory, "*BB*");
+            Assert.Contains(Path.Combine(TestDirectory, "AAABB"), strArr);
             if (Interop.IsWindows)
-            {
-                iCountTestcases++;
-                if (Array.IndexOf(strArr, "aaabbcc") < 0)
-                {
-                    iCountErrors++;
-                    printerr("Error_956yb! Incorrect name==" + strArr[0]);
-                }
-            }
-            iCountTestcases++;
-            if (Array.IndexOf(strArr, "AAABB") < 0)
-            {
-                iCountErrors++;
-                printerr("Error_48yg7! Incorrect name==" + strArr[1]);
-            }
-
-            // [] Should not search on fullpath
-            // [] Search Criteria without match should return empty array
-
-            strLoc = "Loc_2301";
-            FileInfo f = new FileInfo(Path.Combine(dir2.Name, "TestDir1", "Test.tmp"));
-            FileStream fs6 = f.Create();
-            strArr = Directory.GetFileSystemEntries(dir2.Name, Path.Combine("TestDir1", "*"));
-            iCountTestcases++;
-            if (strArr.Length != 1)
-            {
-                iCountErrors++;
-                printerr("Error_28gyb! Incorrect number of files");
-            }
-            fs6.Dispose();
-
-            //if(Directory.Exists(dirName))
-            //	Directory.Delete(dirName, true);
+                Assert.Contains(Path.Combine(TestDirectory, "aaabbcc"), strArr);
+            else
+                Assert.DoesNotContain(Path.Combine(TestDirectory, "aaabbcc"), strArr);
```

Comment:
I'll need to wrangle myself up a mac; I don't currently have one available to use. 

The case sensitivity check was frequently brought up in the tests, so again I'll need to go through them and pull out the OS-specific parts. I'll make a new issue for that and mark it Xplat and work on it as I move forward with redoing the worst of the FileSystem tests.